app/components/job_application_review_component/section.rb

```diff
@@ -44,6 +44,6 @@ def url_helpers
   def allow_edit?
     return @allow_edit unless @allow_edit.nil?
 
-    !@job_application.deadline_passed? && !@job_application.submitted?
+    !@job_application.deadline_passed? && @job_application.draft?
   end
```

Comment:
normally I would look at add/extend a test to cover this new behaviour. Our tests are not in very good shape, so taking one and improving its intentionality would be worth doing IMHO